File: Algorithms/Android/app/src/main/java/com/example/androidalgorithms/testers/SelectSortTester.java
package com.example.androidalgorithms.testers;

import com.example.androidalgorithms.Utils;
import com.example.androidalgorithms.algorithms.SelectSort;

public class SelectSortTester extends AlgorithmTester {

    private int[] arraySizes = { 100, 500, 1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000 };

    @Override
    public String testDescription(int testSize) {
        return String.format("N: %d", arraySizes[testSize]);
    }

    @Override
    public double test(int testSize) {
        double[] arrayToSort = Utils.generateRandomList(arraySizes[testSize]);

        long startTime = System.nanoTime();
        SelectSort.call(arrayToSort);
        return System.nanoTime() - startTime;
    }
}

File: Algorithms/flutter_algorithms/lib/testers/select_sort_tester.dart
import 'package:flutter_algorithms/utils.dart';
import 'package:flutter_algorithms/testers/algorithm_tester.dart';
import 'package:flutter_algorithms/algorithms/select_sort.dart';

class SelectSortTester extends AlgorithmTester {
  final List<int> arraySizes = [100, 500, 1000, 2000, 3000, 4000, 5000, 6000, 7000, 20000];

  @override
  String testDescription(int testSize) {
    return "N: ${arraySizes[testSize]}";
  }

  @override
  int test(int testSize) {
    List<double> arrayToTest = Utils.generateRandomList(arraySizes[testSize]);

    final stopwatch = Stopwatch()..start();
    SelectSort.call(arrayToTest);
    stopwatch.stop();
    return stopwatch.elapsedMilliseconds;
  }
}